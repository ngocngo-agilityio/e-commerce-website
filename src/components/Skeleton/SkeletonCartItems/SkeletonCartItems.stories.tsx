import type { Meta, StoryObj } from '@storybook/react';

// Components
import SkeletonCartItems from '.';

const meta = {
  title: 'Skeletons/SkeletonCartItems',
  component: SkeletonCartItems,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof SkeletonCartItems>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
