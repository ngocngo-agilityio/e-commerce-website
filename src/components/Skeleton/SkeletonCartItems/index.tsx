import { Flex, Skeleton } from '@chakra-ui/react';

const SkeletonCartItems = (): JSX.Element => (
  <>
    <Flex flexDir="column" gap="4px" mt="70px">
      {Array.from({ length: 4 }).map((_, i) => (
        <Skeleton key={i} w="full" h="80px" />
      ))}
    </Flex>

    <Skeleton mt="83px" mb="38px" w="200px" h="50px" />

    <Flex w="531px" mb="41px" flexDir="column" gap="4px">
      {Array.from({ length: 3 }).map((_, i) => (
        <Skeleton key={i} h="54px" w="full" />
      ))}
    </Flex>

    <Skeleton w="262px" h="58px" />
  </>
);

export default SkeletonCartItems;
