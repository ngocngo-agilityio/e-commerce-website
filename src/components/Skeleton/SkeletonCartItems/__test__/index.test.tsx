// Libs
import { render } from '@testing-library/react';

// Components
import SkeletonCartItems from '..';

describe('SkeletonCartItems component', () => {
  test('should match snapshot', () => {
    const { container } = render(<SkeletonCartItems />);

    expect(container).toMatchSnapshot();
  });
});
