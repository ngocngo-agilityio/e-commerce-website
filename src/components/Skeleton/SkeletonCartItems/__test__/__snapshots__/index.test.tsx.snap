// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonCartItems component should match snapshot 1`] = `
<div>
  <div
    class="css-1ndtkor"
  >
    <div
      class="chakra-skeleton css-1w3kri6"
    />
    <div
      class="chakra-skeleton css-1w3kri6"
    />
    <div
      class="chakra-skeleton css-1w3kri6"
    />
    <div
      class="chakra-skeleton css-1w3kri6"
    />
  </div>
  <div
    class="chakra-skeleton css-18h6sy6"
  />
  <div
    class="css-15g4dvl"
  >
    <div
      class="chakra-skeleton css-1nc58od"
    />
    <div
      class="chakra-skeleton css-1nc58od"
    />
    <div
      class="chakra-skeleton css-1nc58od"
    />
  </div>
  <div
    class="chakra-skeleton css-6vl96b"
  />
</div>
`;
